### PR TITLE
Paginate GetSections call

### PR DESCRIPTION
**Pull request by ClaudineL**
The call to retrieve sections is paginated, this ensures that the results won't cut off after 30 entries. I also added a unit test based on the one done for the paginated Tickets call.

### CODE REVIEW

File: src/ZendeskApi_v2/Requests/HelpCenter/Sections.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace ZendeskApi_v2.Requests.HelpCenter
     public interface ISections : ICore
     {
 #if SYNC
-        GroupSectionResponse GetSections();
```
Comment (mozts2005) [inline src/ZendeskApi_v2/Requests/HelpCenter/Sections.cs:13]: please leave the old method and add this as a new overload. adding it as default parameters makes the updated Library binary incompatible for drop in upgrades.

File: test/ZendeskApi_v2.Test/HelpCenter/SectionTests.cs [hunk 2]
```diff
@@ -45,6 +46,46 @@ public void CanGetSections()
             Assert.AreEqual(res1.Section.Id, res.Sections[0].Id.Value);
         }
 
+        [Test]
+        public void CanGetSectionsPaged()
```
Comment (mozts2005) [inline test/ZendeskApi_v2.Test/HelpCenter/SectionTests.cs:50]: Please, Use the ```Assert.That ``` syntax and assert that the deletes complete

Comment (mozts2005) [reply test/ZendeskApi_v2.Test/HelpCenter/SectionTests.cs:50]: Sorry, I Worded my request funny. 

I need all Asserts to use the ```Assert.That``` syntax, not just the deletes

File: src/ZendeskApi_v2/Requests/HelpCenter/Sections.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ namespace ZendeskApi_v2.Requests.HelpCenter
...
+        GroupSectionResponse GetSections(int? perPage = null, int? page = null);
```
Comment (mozts2005) [inline src/ZendeskApi_v2/Requests/HelpCenter/Sections.cs]: Please, Add both the sync and async versions of any new methods

File: src/ZendeskApi_v2/Requests/HelpCenter/Sections.cs [hunk 4]
```diff
@@ -10,7 +10,8 @@ namespace ZendeskApi_v2.Requests.HelpCenter
     public interface ISections : ICore
     {
 #if SYNC
-        GroupSectionResponse GetSections(int? perPage = null, int? page = null);
+        GroupSectionResponse GetSections();
```
Comment (mozts2005) [inline src/ZendeskApi_v2/Requests/HelpCenter/Sections.cs:13]: Please, Add both the sync and async versions of any new methods

File: test/ZendeskApi_v2.Test/HelpCenter/SectionTests.cs [hunk 5]
```diff
@@ -45,6 +46,91 @@ public void CanGetSections()
...
+            var section2 = api.HelpCenter.Sections.CreateSection(new Section {
+                Name = "My Test section 2",
+                Position = 0,
+                CategoryId = category_id
+            });
+
+            const int count = 2;
+            var sectionsAsync = api.HelpCenter.Sections.GetSectionsAsync(count, 1).Result;
+
+            Assert.That(sectionsAsync.Count > 0);
```
Comment (mozts2005) [inline test/ZendeskApi_v2.Test/HelpCenter/SectionTests.cs:110]:
```suggestion
            Assert.That(sectionsAsync.Count, Is.GreaterThan(0));
```

File: test/ZendeskApi_v2.Test/HelpCenter/SectionTests.cs [hunk 6]
```diff
@@ -45,6 +46,91 @@ public void CanGetSections()
...
+
+            Assert.That(count, Is.EqualTo(secondPage.Sections.Count));
+            Assert.That(sectionById2.Section.Id, Is.EqualTo(secondPage.Sections[0].Id.Value));
+
+            var nextPage = secondPage.NextPage.GetQueryStringDict()
+                .Where(x => x.Key == "page")
+                .Select(x => x.Value)
+                .FirstOrDefault();
+
+            Assert.That(nextPage != null);
```
Comment (mozts2005) [inline test/ZendeskApi_v2.Test/HelpCenter/SectionTests.cs:128]:
```suggestion
            Assert.That(nextPage, Is.Not.Null);
```

**Comment by ClaudineL** [comment]
Is it possible to have an ETA for the release of 3.10.7? We'd need those changes fairly soon, especially the addition of the `or_tags` property from my previous PR, as our Zendesk permission model is currently broken without it.

**Comment by mozts2005** [comment]
@ClaudineL Released 3.10.7 today

**Comment by ClaudineL** [comment]
Thank you!

**Pull request merged**